### DISCUSSION TITLE
Providing luarocks support (for other applications)

### DISCUSSION THREAD

**Discussion by takase1121** [keplerproject/luarocks#1379] [Q&A] [answered]
Hi, I'm one of the contributors of [lite-xl][1], a text editor written mostly in Lua. Distributing and managing plugins has always being a problem, so I'm looking to somehow integrate luarocks to the editor, so that:

1. Plugins could utilise rocks like `lfs`
2. lite-xl could load plugins installed via luarocks

Now there's the problems.
1. What are the paths to add to `package.(c)path`?
From what I see in awesomewm, using luarocks is as simple as `require 'luarock.loader'` (don't quote me on the exact name, you get the point). Luarocks is usually installed in `/usr` so that's usually within lua's PATH. How about Windows? What are the common paths that we'll need to add to add luarocks properly?

2. How do we configure luarocks to compile for lite-xl?
lite-xl is usually shipped as a statically compiled program (one binary called `lite-xl` and a directory called `data`). Is it possible to get rocks that needed compilation to compile to that? Or do we ship `liblua.so/dll`? What are the consequences we should be looking for?

lite-xl targets Lua 5.2 and plans to switch to Lua 5.4 in the future. Are there any catches for that?

If I misunderstood something, please let me know. Thank you in advance.

[1]: https://lite-xl.com

**Comment by daurnimator** [comment, marked as answer]
> 1. What are the paths to add to `package.(c)path`?

See `luarocks path --help`.
Note that this works in combination with a `--tree` top argument (and other aliases such as `--global` and `--local`; see `luarocks --help`).


> 2. How do we configure luarocks to compile for lite-xl?

Mostly just make sure luarocks is configured for the correct header location and lua version



> lite-xl targets Lua 5.2 and plans to switch to Lua 5.4 in the future. Are there any catches for that?

See above: you'll need to update the luarocks config (or invocation) when you update. The only real catch I can think of is that users will have to recompile any of their compiled modules.
